### PR TITLE
Add apps to restriction rules in branch protection

### PR DESCRIPTION
**Pull request by jporzucek**
Fixes https://github.com/google/go-github/issues/2506

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,9 @@ require (
 
 require (
 	github.com/golang/protobuf v1.3.2 // indirect
-	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110 // indirect
+	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
```
Comment (gmlewis) [inline go.mod:12]: Please revert all changes to `go.mod` and `go.sum` as they are unrelated to this PR.

File: go.sum [hunk 2]
```diff
@@ -7,21 +7,35 @@ github.com/google/go-cmp v0.5.9/go.mod h1:17dUlkBOakJ0+DkrSSNjCkIjxS6bF9zb3elmeN
 github.com/google/go-querystring v1.1.0 h1:AnCroh3fv4ZBgVIf1Iwtovgjaw/GiKJo8M8yD/fhyJ8=
 github.com/google/go-querystring v1.1.0/go.mod h1:Kcdr2DB4koayq7X8pmAG4sNG59So17icRSOU623lUBU=
 golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2/go.mod h1:djNgcEr1/C05ACkg1iLfiJU5Ep61QUkGW8qpdssI0+w=
+golang.org/x/crypto v0.0.0-20191011191535-87dc89f01550/go.mod h1:yigFU9vqHzYiE8UmvKecakEJjdnWj3jj499lnFckfCI=
```
Comment (gmlewis) [inline go.sum:10]: Please revert all changes to `go.mod` and `go.sum` as they are unrelated to this PR.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @jporzucek .

**Review by gmlewis — APPROVED** [review]
Thank you, @jporzucek !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by raynigon — APPROVED** [review]
LGTM

**Pull request merged**